web/client-api/src/test/java/io/deephaven/web/client/api/AbstractAsyncGwtTestCase.java

```diff
@@ -0,0 +1,391 @@
+package io.deephaven.web.client.api;
+
+import com.google.gwt.core.client.JavaScriptException;
+import com.google.gwt.junit.client.GWTTestCase;
+import elemental2.core.JsArray;
+import elemental2.core.JsError;
+import elemental2.core.JsString;
+import elemental2.dom.CustomEvent;
+import elemental2.dom.CustomEventInit;
+import elemental2.dom.DomGlobal;
+import elemental2.promise.IThenable;
+import elemental2.promise.Promise;
+import io.deephaven.web.client.api.subscription.ViewportData;
+import io.deephaven.web.client.fu.CancellablePromise;
+import io.deephaven.web.client.ide.IdeSession;
+import io.deephaven.web.shared.data.Viewport;
+import io.deephaven.web.shared.fu.JsRunnable;
+import io.deephaven.web.shared.fu.RemoverFn;
+import jsinterop.annotations.JsMethod;
+import jsinterop.annotations.JsPackage;
+import jsinterop.annotations.JsProperty;
+import jsinterop.base.Js;
+import jsinterop.base.JsPropertyMap;
+import org.apache.tapestry.INamespace;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.BooleanSupplier;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import static elemental2.dom.DomGlobal.console;
+
+public abstract class AbstractAsyncGwtTestCase extends GWTTestCase {
+    @JsMethod(namespace = JsPackage.GLOBAL)
+    private static native Object eval(String code);
+
+    private static Promise<JsPropertyMap<Object>> importScript(String moduleName) {
+        return (Promise<JsPropertyMap<Object>>) eval("import('" + moduleName + "')");
+    }
+
+    private static Promise<Void> importDhInternal() {
+        return importScript(localServer + "/jsapi/dh-internal.js")
+                .then(module -> {
+                    Js.asPropertyMap(DomGlobal.window).set("dhinternal", module.get("dhinternal"));
+                    return Promise.resolve((Void) null);
+                });
+    }
+
+    public static final String localServer = System.getProperty("dh.server", "http://localhost:10000");
+
+    public static class TableSourceBuilder {
+        private final List<String> pythonScripts = new ArrayList<>();
+
+        public TableSourceBuilder script(String script) {
+            pythonScripts.add(script);
+            return this;
+        }
+
+        public TableSourceBuilder script(String tableName, String python) {
+            pythonScripts.add(tableName + "=" + python);
+
+            return this;
+        }
+    }
+
+    /**
+     * Set this to a value higher than 1 to get more time to run debugger without timeouts failing.
+     */
+    protected static final int TIMEOUT_SCALE = 2;
```

Comment:
Feel like this should default to `1`, particularly with the comment before it. Should it default to a different value if we're in a dev mode vs. CI (unsure which flag that is)? Or some other env var?

Reply:
Yep, I did intend to revert this... and tweak all the tests to have massively larger timeouts. Will clean that up.